docs/animation/asm.md

```diff
@@ -0,0 +1,70 @@
+ASM Files
+===========
+
+ASM Files are the meat of the animation API. These define how the animation is carried out and how to use the clips defined in the armature file.
+
+Concepts
+----------
+
+The ASM contains _parameters_, _clips_, _states_, and _transitions_. The simplest ones to explain are states and transitions, so I'll start with those
+
+### States
+
+The Animation _State_ Machine can be in many different _states_. You define which states there are in the states section.
+Each _clip_ implements a state, telling the ASM how to animate it.
+
+### Transitions
+
+Right now, the only uses for transitions are to define which states can go to which other states, no way to make a transition-only
+clip exists. In the code, the only uses for the `transitions` member of AnimationStateMachine is to verify that you said a transition
+can occur between the current state and the desired next state.
+
+### Parameters
+
+!!! note
+    
+    Parameters are called `TimeValues` in the code
+
+Each parameter can either be defined in the ASM or at load-time. Load-time parameters are usually of the type `VariableValue`, allowing you to set them from within your code.
+Other kinds are `SimpleExprValue`, allowing you to do calculations relative to the current game time, `CompositionValue`, allowing you to chain parameters together and `ParameterValue`, allowing you to reference other parameters.
+A parameter is an input to a clip, telling it how far into the animation to animate.
+
+
+### Clips
+
+!!! note
+    
+    Clips can either be ASM-clips, ones that are defined in the ASM, or armature-clips, ones that are defined in the armature file.
+    For the rest of this page, "clips" will refer to ASM-clips unless otherwise stated
+
+A clip controls what a state will do. They can be one of many types, the easiest to use being the `TimeClip`. These animate a given clip by a given parameter.
+Other kinds include the `IdentityClip`, which does nothing, the `ClipReference`, which should "only be used for debugging" and just does the same as another clip,
```

Comment:
when writing documentation, "idk" is not a good thing to be saying ;p

I think this is the code-side deserialization of a `#` reference in the ASM json, that's what the comment means. So it's not that you should only use it for debugging, it's just that you'll find it rare to construct one in code instead of specifying it in json